src/runmode-dpdk.c

```diff
@@ -0,0 +1,1295 @@
+    }
+    retval = ConfigSetChecksumChecks(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueBoolWithDefault(
+                if_root, if_default, dpdk_yaml.checksum_checks_offload, &entry_bool) != 1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing",
+                dpdk_yaml.checksum_checks_offload);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetChecksumOffload(iconf, entry_bool);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_mode, &copy_mode_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_mode);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    if (ConfGetChildValueWithDefault(if_root, if_default, dpdk_yaml.copy_iface, &copy_iface_str) !=
+            1) {
+        SCLogError(SC_ERR_DPDK_CONF, "Value of \"%s\" invalid or missing", dpdk_yaml.copy_iface);
+        errno = EINVAL;
+        SCReturnInt(-errno);
+    }
+    retval = ConfigSetCopyIfaceSettings(iconf, copy_iface_str, copy_mode_str);
+    if (retval < 0) {
+        return retval;
+    }
+
+    if (iconf->copy_mode == DPDK_COPY_MODE_NONE) {
+        SCLogInfo("IDS mode enabled, disabling TX settings");
+        iconf->nb_tx_queues = 0;
+        iconf->nb_tx_desc = 0;
+    }
+
+    SCReturnInt(0);
+}
+DPDKIfaceConfig *ConfigParse(const char *iface)
+{
+    int retval;
+    DPDKIfaceConfig *iconf = NULL;
+    if (iface == NULL) {
+        SCLogError(SC_ERR_DPDK_CONF, "Provided iface is NULL");
+        SCReturnPtr(NULL, "void *");
+    }
+
+    ConfigInit(&iconf);
+    if (iconf == NULL) {
+        SCReturnPtr(NULL, "void *");
+    }
+    retval = ConfigLoad(iconf, iface);
+    if (retval < 0) {
+        iconf->DerefFunc(iconf);
+        SCReturnPtr(NULL, "void *");
+    }
+
+    return iconf;
+}
+
+void DeviceInitPortConf(const DPDKIfaceConfig *iconf, const struct rte_eth_dev_info *dev_info,
+        struct rte_eth_conf *port_conf)
+{
+    *port_conf = (struct rte_eth_conf){
+            .rxmode = {
+                    .mq_mode = ETH_MQ_RX_NONE,
+                    .max_rx_pkt_len = RTE_ETHER_MAX_LEN,
+                    .offloads = 0, // turn every offload off to prevent any packet modification
+            },
+            .txmode = {
+                    .mq_mode = ETH_MQ_TX_NONE,
+                    .offloads = 0,
+            },
+    };
+
+    // configure RX offloads
+    if (dev_info->rx_offload_capa & DEV_RX_OFFLOAD_RSS_HASH) {
+        if (iconf->nb_rx_queues > 1) {
+            SCLogConfig("RSS enabled on %s for %d queues", iconf->iface, iconf->nb_rx_queues);
+            port_conf->rx_adv_conf.rss_conf.rss_key = rss_hkey;
+            port_conf->rx_adv_conf.rss_conf.rss_key_len = RSS_HKEY_LEN;
+            port_conf->rx_adv_conf.rss_conf.rss_hf = ETH_RSS_TCP | ETH_RSS_UDP;
```

Comment:
in v3 and v4 I had raised a concern in case non first fragment the is no L4 headers (TCP, UDP and SCTP) almost all nic supporting RSS does skip for non first fragments in RSS calculation. Hence requested to keep 3 tuple as default. Please rework this